src/breakupModel/model/Satellite.h

```diff
@@ -279,6 +294,33 @@ class Satellite {
         _velocity = velocity;
     }
 
+    [[nodiscard]] const std::array<double, 3> &getEjectionVelocity() const {
+        return _ejectionVelocity;
+    }
+
+    /**
+     * Sets the ejection velocity of this Satellite (debris-fragment).
+     * @note This does modifies the state, but the the ejection velocity is property of the breakup and not directly
+     * linked to the Satellite like it is the "whole" velocity. The _orbitalElementsCache is therefore not invalidated.
+     * @param ejectionVelocity - cartesian vector
+     */
+    void setEjectionVelocity(const std::array<double, 3> &ejectionVelocity) {
+        _ejectionVelocity = ejectionVelocity;
+    }
+
+    /**
+     * Sets the ejection velocity of this Satellite (debris-fragment)
+     * and adds in on top of the base velocity.
+     * @note This does modifies the state, but the the ejection velocity is property of the breakup and not directly
+     * linked to the Satellite like it is the "whole" velocity. The _orbitalElementsCache is therefore not invalidated.
+     * @param ejectionVelocity - cartesian vector
+     */
+    void addEjectionVelocity(const std::array<double, 3> &ejectionVelocity) {
+        using util::operator+;
+        _ejectionVelocity = ejectionVelocity;
+        _velocity = _velocity + _ejectionVelocity;
```

Comment:
This seems a little dangerous. What happens if for some reason _velocity was not initialized?

Reply:
Just wondering if it wasn't better if _velocity and _ejectionVelocity get initiliazed in the same function to avoid any possible scenarios or that happening in the wrong order etc.?

Reply:
At the moment, both of them are standardise zero initialized, but i agree that the current method is not that "optimal".

Nevertheless, I would leave it as it is for the moment because those things will undergo a rework with the Data Oriented Approach.

Reply:
Just got the idea for the Data Oriented Approach:
I will remove the ejection velocity attribute of Satellite because it's just a property of a breakup. It makes no sense to waste memory for "ejection velocity" for input satellites.

The "Satellite**s**" class on the other hand will contain a possibility to get the ejection velocity because this structure will be used for the output